ThunderstoreCLI/Utils/DownloadCache.cs

```diff
@@ -0,0 +1,53 @@
+namespace ThunderstoreCLI.Utils;
+
+public sealed class DownloadCache
+{
+    public string CacheDirectory { get; }
+
+    private HttpClient Client { get; } = new()
+    {
+        Timeout = TimeSpan.FromHours(1)
+    };
+
+    public DownloadCache(string cacheDirectory)
+    {
+        CacheDirectory = cacheDirectory;
+    }
+
+    // Task instead of ValueTask here because these Tasks will be await'd multiple times (ValueTask does not allow that)
+    public Task<string> GetFileOrDownload(string filename, string downloadUrl)
+    {
+        string fullPath = Path.Combine(CacheDirectory, filename);
+        if (File.Exists(fullPath))
+        {
+            return Task.FromResult(fullPath);
+        }
+
+        return DownloadFile(fullPath, downloadUrl);
+    }
+
+    private async Task<string> DownloadFile(string fullpath, string downloadUrl)
+    {
+        int tryCount = 0;
+Retry:
+        try
+        {
+            var tempPath = fullpath + ".tmp";
```

Comment:
Should all this stuff really be done in a single `try` block? E.g. if `File.Move` fails constantly for some reason, we end up downloading the file file times for no reason?

Reply:
Moved out the File.Move, good catch